### PR TITLE
Oven Mode XML

### PR DESCRIPTION
**Pull request by abeck-whirlpool**
Adding the new XML for the Oven Mode cluster
Competed Regen of files with new XML and enabled via ZAP

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="ModeTag" type="enum16">
+    <cluster code="0x0049"/>
+    <item value="0x4000" name="Bake"/>
+    <item value="0x4001" name="Convection"/>
+    <item value="0x4002" name="Grill"/>
+    <item value="0x4003" name="Roast"/>
+		<item value="0x4004" name="Clean"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:26]: Why the weird indent here and for Proofiing?

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  <cluster>
+    <domain>Appliances</domain>
+    <name>Oven Mode</name>
+    <code>0x0049</code>
+    <define>OVEN_MODE_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>Attributes and commands for selecting a mode from a list of supported options.</description>
+    <globalAttribute side="either" code="0xFFFD" value="2"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:41]: This does not match the spec.

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    <!-- Base data types -->
+    <attribute side="server" code="0x0000" define="SUPPORTED_MODES"  type="ARRAY" entryType="ModeOptionStruct" writable="false" optional="false" isNullable="false" length="255">SupportedModes</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:43]: ModeOptionStruct needs to have this cluster added to its list of clusters, I assume?

File: src/app/zap_cluster_list.json [hunk 4]
```diff
@@ -201,6 +202,7 @@
         "OTA_BOOTLOAD_CLUSTER": [],
         "OTA_SOFTWARE_UPDATE_PROVIDER_CLUSTER": ["ota-provider"],
         "OTA_SOFTWARE_UPDATE_REQUESTOR_CLUSTER": ["ota-requestor"],
+        "OVEN_MODE_CLUSTER": ["oven-mode-server"],
```
Comment (bzbarsky-apple) [inline src/app/zap_cluster_list.json:205]: There is no oven-mode-server...

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml (hunk 1 above)
Comment (abeck-whirlpool) [reply src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:26]: I'm not seeing the padding locally even on different text editors. Will try and figure that formatting out and get it fixed

File: src/app/zap_cluster_list.json (hunk 4 above)
Comment (abeck-whirlpool) [reply src/app/zap_cluster_list.json]: Fixed! Got carried away and forgot to list it as mode-base-server thanks for catching that!

File: data_model/clusters/Mode_Oven.xml [hunk 5]
```diff
@@ -54,10 +54,38 @@ This notice and disclaimer must be included on all copies of this document.
 Connectivity Standards Alliance
 508 Second Street, Suite 206
 Davis, CA 95616, USA
--->
-<cluster xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="types types.xsd cluster cluster.xsd" id="0x0049" name="Oven Mode" revision="1">
-  <revisionHistory>
-    <revision revision="1" summary="Initial Release"/>
-  </revisionHistory>
-  <classification hierarchy="derived" baseCluster="Mode Base" role="application" picsCode="OTCCM" scope="Endpoint"/>
+-->
```
[the remark this answers is not in the document]
Comment (abeck-whirlpool) [reply data_model/clusters/Mode_Oven.xml:57]: Yeah I misunderstood what was needed for the derivative. Just removed and did a regen again to be safe but shouldn't have this manually in there any longer

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="ModeTag" type="enum16">
+    <cluster code="0x0049"/>
+    <item value="0x4000" name="Bake"/>
+    <item value="0x4001" name="Convection"/>
+    <item value="0x4002" name="Grill"/>
+    <item value="0x4003" name="Roast"/>
+		<item value="0x4004" name="Clean"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:26]: You need to find an editor that makes tabs-vs-spaces clear.
```suggestion
    <item value="0x4003" name="Roast"/>
    <item value="0x4004" name="Clean"/>
```

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    <item value="0x4005" name="Convection Bake"/>
+    <item value="0x4006" name="Convection Roast"/>
+    <item value="0x4007" name="Warming"/>
+		<item value="0x4008" name="Proofing"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:30]:
```suggestion
    <item value="0x4007" name="Warming"/>
    <item value="0x4008" name="Proofing"/>
```

File: src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  </enum>
+
+  <dataTypes>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/oven-mode-cluster.xml:33]: What is this and where is it coming from?  This is not something that happens in ZAP XML.

**Review by woody-apple — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**